### ISSUE TITLE
Nullable response have a weird behavior right now 

### ISSUE THREAD

**Issue by timotheeguerin** [microsoft/cadl#356]
```cadl
import "@cadl-lang/rest";

using Cadl.Http;

@error
model Error {
  message: string;
}

@route("/1") op read1(): string | null;
@route("/2") op read3(): string | null | Error;
```

[Playground link](https://cadlplayground.z22.web.core.windows.net/?c=aW1wb3J0ICJAY2FkbC1sYW5nL3Jlc3QiOwoKdXNpbmcgQ2FkbC5IdHRwOwoKQGVycm9yCm1vZGVsIEXEDCB7CiAgbWVzc2FnZTogc3RyaW5nOwp9CgpAcm91dGUoIi8xIikgb3AgcmVhZDEoKcgkIHwgbnVsbDvKKDLKKDPRKCB8xm87Cg%3D%3D)

**Comment by mikekistler** [comment]
I agree with @timotheeguerin that a return type of `string | null` should result in a response schema:
```
    "schema": {
      "type": "string",
      "nullable": true
    }
```

**Comment by timotheeguerin** [comment]
- `null` present in op return union means each response is nullable?

**Comment by nguerrera** [comment]
Asking for re-triaqge here. As soon as someone mentions the word "null" in a meeting, we go down a rabbit hole and it's hurtful when our code doesn't actually interpret "null" the way it should be interpreted.
